### PR TITLE
Add Gaia photometric alert catalog

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR makes it possible to ingest the Gaia photometric alert catalog.

### CODE REVIEW

File: skyportal/handlers/api/catalog_services.py [hunk 1]
```diff
@@ -460,3 +471,206 @@ def fetch_swift_transients(instrument_id, user_id):
...
+            Repeated posting will skip the existing source.
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  telescope_name:
+                    required: false
+                    type: integer
```
Comment (guynir42) [inline skyportal/handlers/api/catalog_services.py:492]:
```suggestion
                    type: string
```

File: skyportal/handlers/api/catalog_services.py [hunk 2]
```diff
@@ -460,3 +471,206 @@ def fetch_swift_transients(instrument_id, user_id):
...
+                    description: |
+                      Name of telescope to assign this catalog to.
+                      Use the same name as your nickname
+                      for Gaia. Defaults to Gaia.
+                  startDate:
+                    required: false
+                    type: float
```
Comment (guynir42) [inline skyportal/handlers/api/catalog_services.py:499]: Should these be arrow parsable strings?

File: skyportal/handlers/api/catalog_services.py [hunk 3]
```diff
@@ -460,3 +471,206 @@ def fetch_swift_transients(instrument_id, user_id):
...
+    payload : dict
+        Dictionary containing filtering parameters
+    """
+
+    session = Session()
+    obj_ids = []
+
+    lightcurve_url = "https://gsaweb.ast.cam.ac.uk/alerts/alert"
+    alert_url = "http://gsaweb.ast.cam.ac.uk/alerts/alerts.csv"
+    # alert_url = "./alerts.csv"
```
Comment (guynir42) [inline skyportal/handlers/api/catalog_services.py:566]:
```suggestion
```

File: skyportal/handlers/api/catalog_services.py [hunk 4]
```diff
@@ -460,3 +471,206 @@ def fetch_swift_transients(instrument_id, user_id):
...
+
+        while not file_read and nretries < 10:
+            try:
+                table = Table.read(alert_url, format='csv')
+                file_read = True
+            except FileNotFoundError:
+                nretries = nretries + 1
+                time.sleep(10)
+        if not file_read:
+            log('Failed to read Gaia catalog')
```
Comment (guynir42) [inline skyportal/handlers/api/catalog_services.py:580]:
```suggestion
            log('Failed to read Gaia alert catalog')
```

File: skyportal/handlers/api/catalog_services.py [hunk 5]
```diff
@@ -460,3 +471,206 @@ def fetch_swift_transients(instrument_id, user_id):
...
+                if date > end_date:
+                    continue
+
+            data = {'ra': ra, 'dec': dec, 'id': name}
+            s = session.scalars(Obj.select(user).where(Obj.id == name)).first()
+            if s is None:
+                obj_id = post_source(data, user_id, session)
+                obj_ids.append(obj_id)
+            else:
+                obj_id = s.id
```
Comment (guynir42) [inline skyportal/handlers/api/catalog_services.py:607]: Are you sure you don't want to append to `obj_ids` in case where the source is found?

**Review by guynir42 — APPROVED** [review]
Some minor comments below:

File: skyportal/handlers/api/catalog_services.py (hunk 5 above)
Comment (mcoughlin) [reply skyportal/handlers/api/catalog_services.py:606]: I return those obj_ids that get posted to display on the front-end that they were added.

File: skyportal/handlers/api/catalog_services.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/catalog_services.py]: Good point.

**Pull request merged**